bin/taste-tester

```diff
@@ -322,10 +323,15 @@ MODES:
     end
 
     opts.on(
-      '-s', '--servers SERVER[,SERVER]', Array,
-      'Server to test/untest/keeptesting.'
-    ) do |s|
-      options[:servers] = s
+      '-s', '--split', 'Split server list on commas.'
+    ) do
+      options[:split] = true
+    end
```

Comment:
There is no need for such an option. It should certainly just handle the case in which the first argv is coma-separated, or space separated, or some combination thereof. Neither character is allowed in hostnames, so the intended behavior is very clear.

What i don't want is to be passing in sometimes with -s arg and sometimes with positional arguments.

There are two acceptable options here:

1. Accept `-` to `--servers` (preferred) - and actually is better for you since what you're doing is piping in from other commands.
2. Migrate to positional arguments, we can keep `-s` around as a no-op for backcompat.

In both cases `split(/[\s,]+/)` so people can pass in from whatever the fuck they want.

Again, making an option for stdin seems *significantly* better all around.

The only thing that shouldn't happen is different parsing for different ways of passing it in. That's the worst UI possible.

Reply:
I think positional arguments are a lot more common. I can even implement a `-` as argument which would read stdin.
But positional arguments feel much logical to me. It's like` grep pattern file1 file2` Or more `systemctl status something somethingelse`
The main problem with stdin is how do you deal with user input?
Like, `are you sure you want to do this?`, or `Do you want to amend your commit message?`
You can't really read the list of hosts from stdin and then read user input
You could read from a file and use `<( ... )` but that is cumbersome

I personally vote for deprecating `-s` entirely, meaning keeping it as a noop for backward compatibility